demos/text_detection_demo/cpp_gapi/src/text_recognition.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "text_recognition.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <numeric>
+#include <fstream>
+
+bool fileExists(const std::string& filename) {
+    std::ifstream f(filename.c_str());
+    return f.good();
+}
+bool tryReadVocabFile(const std::string& filename, std::string& alphabet) {
+    if (fileExists(filename)) {
+        alphabet = "";
+        std::ifstream inputFile(filename);
+        if (!inputFile.is_open()) {
+            throw std::runtime_error("Can't open the vocab file: " + filename);
+        }
+        std::string vocabLine;
+        while (std::getline(inputFile, vocabLine)) {
+            if (vocabLine.length() != 1) {
+                throw std::invalid_argument("Lines in the vocab file must contain 1 character");
+            }
+            alphabet += vocabLine;
+        }
+        if (alphabet.empty()) {
+            throw std::logic_error("File is empty: " + filename);
+        }
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void softmaxAndChoose(const std::vector<float>::const_iterator& begin,
+                      const std::vector<float>::const_iterator& end,
+                      int *argmax, float *prob) {
+    auto maxElem = std::max_element(begin, end);
+    *argmax = static_cast<int>(std::distance(begin, maxElem));
+    float maxVal = *maxElem;
+    double sum = 0;
+    for (auto i = begin; i != end; i++) {
+        sum += std::exp((*i) - maxVal);
+    }
+    if (std::fabs(sum) < std::numeric_limits<double>::epsilon()) {
+        throw std::logic_error("sum can't be equal to zero");
+    }
+    *prob = 1.0f / static_cast<float>(sum);
+}
+
+std::vector<float> softmax(const std::vector<float>::const_iterator& begin,
```

Comment:
You already have softmax. Can you use version from custom_kernels.cpp?

Reply:
I thought about that too - but they.. are different, for some reasons. One is "softmaxing" pairs in a vector, the other is doing it with all the vector values

But I think I can unite them, or, specifically speaking, use this function in the function of `custom_kernels`